src/core_io.h

```diff
@@ -44,8 +44,8 @@ UniValue ValueFromAmount(const CAmount& amount);
 std::string FormatScript(const CScript& script);
 std::string EncodeHexTx(const CTransaction& tx, const int serializeFlags = 0);
 std::string SighashToStr(unsigned char sighash_type);
-void ScriptPubKeyToUniv(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex);
+void ScriptPubKeyToUniv(const CScript& scriptPubKey, UniValue& out, bool fIncludeHex, bool include_addresses);
 void ScriptToUniv(const CScript& script, UniValue& out, bool include_address);
-void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry, bool include_hex = true, int serialize_flags = 0, const CTxUndo* txundo = nullptr);
+void TxToUniv(const CTransaction& tx, const uint256& hashBlock, UniValue& entry, bool include_addresses, bool include_hex = true, int serialize_flags = 0, const CTxUndo* txundo = nullptr);
```

Comment:
e2b814e315f011b6afea26a1ac10aa9f31b5fddb:

This is not safe against silent merge conflicts. Passing an array of bool args (some with default values) without named args can lead to bugs. I think there are two options:

* Remove all default args (might be controversial)
* Change the order so that no two neighbouring args have the same type

I think you can just add the new arg to the beginning (or close to it). It will be removed anyway, so any style concerns that the arg is "too early" are irrelevant.

Reply:
I updated `TxToUniv` to use this wrapper approach, so it is now used in both cases. As to why I went with this wrapper approach in the first place, it was suggested earlier in the review (unless I mis-interpreted something):

`I'd suggest just using IsDeprecatedRPCEnabled, in the top-level RPC code, and pass down a boolean to wherever it's needed (e.g. in core_write, make ScriptPubKeyToUniv take a boolean to indicate whether to not to include reqSigs / addresses).`

I know it reduces the diff a bit and meant we didn't have changes in as many files. I *think* it got rid of a circular include dependency as well (since we don't have to include `rpc/server` in a bunch of places now). -- and it may have even been necessary to allow this PR to happen without touching the build system as including `rpc/server` in some files was causing trouble. I don't entirely remember, but know it was done for sane reasons, and I think you had suggested something like this early in the review (although your earlier review comments aren't showing in this PR now)

Reply:
I meant for this to be a response to your other comment:
"Can you explain why you are using this wrapper approach for ScriptPubKeyToUniv but not for TxToUniv. Either use it for both or none."